lnclient/lnd/lnd.go

```diff
@@ -492,6 +503,60 @@ func (svc *LNDService) MakeInvoice(ctx context.Context, amount int64, descriptio
 	return transaction, nil
 }
 
+func (svc *LNDService) getRoutingHints(ctx context.Context) (hints []*lnrpc.RouteHint, err error) {
```

Comment:
Did you take inspiration from anywhere for this code? I think there are quite a few issues here.

Can you please check the LND codebase and some other lightning wallets (e.g. https://github1s.com/ZeusLN/zeus/blob/HEAD/lndmobile/index.ts#L632-L633)

I think there's no limit right now, right? we need to limit it to 3 (the "best" options based on incoming liquidity and fees)

Can you also check if we have 1 peer the route hint is _exactly_ the same as is when not using this code. We need to make very sure we don't break this code otherwise no-one with private channels will be able to receive.

Reply:
I took inspiration from Breez. I will also check the Zeus code and see if we can do any improvements.

> I think there's no limit right now, right? we need to limit it to 3 (the "best" options based on incoming liquidity and fees)

LND has a max limit of 20 hints, but yeah will limit this to use best 3 hints. I think the only parameter to decide what's "better" would be incoming liquidity right?

> Can you also check if we have 1 peer the route hint is exactly the same as is when not using this code.

Yup, I checked that and it matches in my local Polar setup, will make the above changes and check again to be sure :+1: